frankenphp.go

```diff
@@ -628,9 +628,100 @@ func go_read_cookies(threadIndex C.uintptr_t) *C.char {
 
 //export go_log
 func go_log(threadIndex C.uintptr_t, message *C.char, level C.int) {
-	ctx := phpThreads[threadIndex].context()
 	m := C.GoString(message)
+	phpLog(phpThreads[threadIndex].context(), m, level)
+}
+
+//export go_log_n
+func go_log_n(threadIndex C.uintptr_t, message *C.char, len C.int, level C.int) {
+	m := C.GoStringN(message, len)
+	phpLog(phpThreads[threadIndex].context(), m, level)
+}
+
+//export go_log_attrs
+func go_log_attrs(threadIndex C.uintptr_t, message *C.char, len C.int, level C.int, cattrs *C.zval) {
+	m := C.GoStringN(message, len)
+
+	var lvl slog.Level
+	if level < -8 || level > 8 {
+		lvl = slog.LevelInfo
+	} else {
+		lvl = slog.Level(level)
+	}
```

Comment:
```suggestion
	lvl = slog.Level(level)
```

According to the docs, any int can be used as level: https://pkg.go.dev/log/slog#hdr-Levels